lib/index.coffee

```diff
@@ -51,6 +53,29 @@ streamToString = (stream) ->
 		.on 'end', ->
 			resolve(Buffer.concat(chunks).toString())
 
+setupMetrics = (metrics) ->
+	metrics.describe('gc_space_reclaimed', 'Disk space the GC must free')
+	metrics.describe('gc_number_images_to_remove', 'Number of images to be removed at a given time')
+	metrics.describe('gc_duration_milliseconds', 'Milliseconds taken to free GC the reclaimed space')
+	metrics.describe('gc_image_removal_errors',
+		'Number of image removal errors by type',
+		{
+			buckets: [ 404, 409, 500 ]
+		}
+	)
+
+	metrics.describe(
```

Comment:
same as above, should be `metrics.describe.counter` and `labelNames` instead of `buckets`

Reply:
:+1:

Reply:
Same as above, I would like to have metrics per removal type. Currently I observe like this:

```
@metrics?.histogram('gc_images_removed', removalType)
```
should I observe it like this?

```
@metrics?.inc('gc_images_removed', 1, removalType)
```

Reply:
Labels are used for this, you can have a single metric name which records numerous timeseries. A single timeries is `metric name + label key-values`, for example all of these are distinct time series:

```
http_requests_completed{statusCode="200"}
http_requests_completed{statusCode="404"}
http_requests_completed{statusCode="500"}
http_requests_completed{statusCode="500", userAgent="chrome"}
http_requests_completed{statusCode="500", userAgent="safari"}
```

Reply:
So, you would observe like,

```{coffeescript}
@metrics?.inc('gc_images_removed', 1, {removalType: removalType})
```

or (not sure if coffeescript supports this) you could remove the duplication (I know this works in newer versions of JS, and typescript)

```{coffeescript}
@metrics?.inc('gc_images_removed', 1, {removalType})
```